cmake/modules/AddSwift.cmake

```diff
@@ -1538,9 +1538,12 @@ function(add_swift_library name)
                    "-latomic")
             # the same issue on FreeBSD, missing symbols:
             # __atomic_store, __atomic_compare_exchange, __atomic_load
+            # NOTE: the "-l" is required here, otherwise Ninja will fail
+            # on clean builds because the library might not have been copied
+            # yet.
             elseif("${sdk}" STREQUAL "FREEBSD")
               list(APPEND swiftlib_private_link_libraries_targets
-                   "${SWIFTLIB_DIR}/clang/lib/freebsd/libclang_rt.builtins-${arch}.a")
+                   "-l${SWIFTLIB_DIR}/clang/lib/freebsd/libclang_rt.builtins-${arch}.a")
```

Comment:
I'll defer this to somebody else, I'm not so sure why you need the `-l`

Reply:
CMake's `target_link_libraries` or `*_LINK_LIBRARIES` properties will interpret input as either a target or as a file. Since targets can't have slashes in them, it can only be a file. The stdlib will now have a dependency on this file.

When a Ninja build starts it will check for dependencies and see that there's a file missing and no target that will export it. This will trigger an error. 

I believe there is a target somewhere else in the CMake script that will copy this file once the build starts. I couldn't find it, but could see that if this line was commented out the file would "magically" appear and subsequent builds were fine even without the `-l`. However the copy target never declared that this specific file will be its output causing Ninja to fail due to a missing dependency and not knowing how to create it.

Ideally this should be solved by fixing the copy target outputs but the link directive is a workaround anyway. Or (probably?) fixed in the compiler by properly linking the library whenever the compiler emits code using atomics.

Reply:
Ugh, there is a task open to have the driver add this, this is quite unfortunate.

Reply:
I also encountered the same problem.Linking it manually make it work temporarily.